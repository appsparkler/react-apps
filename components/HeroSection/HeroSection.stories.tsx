import type { Meta, StoryObj } from "@storybook/react";
import { HeroSection } from "./HeroSection";
// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta: Meta<typeof HeroSection> = {
  title: "Components/Hero Section",
  component: HeroSection,
  tags: ["autodocs"],
  argTypes: {},
};
export default meta;
type Story = StoryObj<typeof HeroSection>;
// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const HeroSectionStory: Story = {
  args: {},
};
HeroSectionStory.storyName = "Hero Section";
