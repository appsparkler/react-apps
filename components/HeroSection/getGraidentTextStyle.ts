import { SxProps, Theme } from "@mui/material";

export function getGraidentTextStyle(
  color1: string,
  color2: string,
  direction: "to right" | "to left" = "to right"
): SxProps<Theme> {
  return {
    background: `linear-gradient(${direction}, ${color1}, ${color2})`,
    WebkitBackgroundClip: "text",
    WebkitTextFillColor: "transparent",
  };
}
