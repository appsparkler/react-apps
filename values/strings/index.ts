export const stringResources = {
  name: "Aakash Shah",
  mobile: "+91-733-8080-855",
  email: "appsparkler@gmail.com",
  social: "appsparkler",
  title: "React & Android Developer",
};
