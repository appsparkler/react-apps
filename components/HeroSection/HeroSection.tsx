import { stringResources } from "@/values/strings";
import { Box, Typography } from "@mui/material";
import { getGraidentTextStyle } from "./getGraidentTextStyle";
import { AndroidSvg } from "../Icons/AndroidSvg";

export const HeroSection = () => {
  return (
    <Box display={"flex"} flexDirection={"column"} alignItems={"center"}>
      <AndroidSvg
        aria-label="android image logo"
        sx={{ height: 75, width: 180 }}
      />
      <Typography variant="h1">{stringResources.name}</Typography>
      <Box
        sx={(theme) => ({
          ...getGraidentTextStyle(
            theme.palette.secondary.main,
            theme.palette.primary.main
          ),
        })}
      >
        <Typography variant="subtitle1" sx={{}}>
          {"React & Android Developer"}
        </Typography>
      </Box>
    </Box>
  );
};
