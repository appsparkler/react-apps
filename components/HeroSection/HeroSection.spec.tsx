import { render, screen } from "@testing-library/react";
import { HeroSection } from "./HeroSection";
import { stringResources } from "@/values/strings";

describe("HeroSection Snapshots", () => {
  it("should match snapshots", () => {
    const { container } = render(<HeroSection />);
    expect(container).toMatchSnapshot();
  });
});

describe("HeroSection Functional Tests", () => {
  it("should render name and title", () => {
    render(<HeroSection />);
    expect(screen.getByText(stringResources.name)).toBeVisible();
    expect(screen.getByText(stringResources.title)).toBeVisible();
    expect(screen.getByLabelText("android logo")).toBeVisible();
    expect(screen.getByLabelText("react logo")).toBeVisible();
  });
});
