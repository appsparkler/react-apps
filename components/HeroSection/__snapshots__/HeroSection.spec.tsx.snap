// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection Snapshots should match snapshots 1`] = `
<div>
  <div
    class="MuiBox-root css-dvxtzn"
  >
    <svg
      aria-hidden="true"
      aria-label="android image logo"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4aik5i-MuiSvgIcon-root"
      fill="none"
      focusable="false"
      height="31"
      viewBox="0 0 138 31"
      width="138"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M68.1553 1.17894C68.7859 1.1085 69.5669 0.821507 70.0969 1.31982C70.1924 1.99816 70.1522 2.68693 70.1572 3.37048C70.1396 12.0662 70.1698 20.7619 70.1419 29.4577C69.494 29.6844 68.8158 29.6951 68.1598 29.4994C68.1651 28.0619 68.1674 26.6241 68.1573 25.1868C66.5948 27.3418 64.3791 29.0246 61.8394 29.6742C58.7319 30.4699 55.2705 29.7887 52.7355 27.7305C51.0451 26.4884 49.8092 24.6595 49.0503 22.6743C47.8771 19.4314 48.2112 15.6168 50.0199 12.6924C51.5222 10.2921 53.866 8.4032 56.5815 7.764C59.4378 7.09089 62.598 7.48745 65.0649 9.1885C66.2783 9.96337 67.2505 11.0618 68.1651 12.181C68.1528 8.51278 68.1757 4.84455 68.1553 1.17894ZM57.2151 9.78596C52.9621 10.7695 49.8172 15.364 50.4779 19.8618C50.8647 22.9926 52.9345 25.7164 55.6151 27.1148C57.8833 28.0384 60.5838 28.2184 62.7844 26.9922C66.4071 25.4894 68.6505 21.1609 67.9169 17.1955C67.4899 14.443 65.7764 11.9436 63.3976 10.6469C61.5409 9.54593 59.2851 9.33461 57.2151 9.78596ZM134.05 1.312C134.585 0.771937 135.369 1.06414 136 1.22068C135.985 10.6495 135.995 20.076 135.995 29.5046C135.334 29.7107 134.653 29.6873 134.015 29.4211C134.003 28.023 134.018 26.622 133.993 25.2233C132.114 27.8558 129.167 29.8412 125.956 29.9795C123.831 30.1175 121.696 29.5513 119.814 28.539C117.109 26.9896 115.091 24.182 114.491 21.0252C113.943 18.2205 114.401 15.1863 115.88 12.7599C117.43 10.1455 120.045 8.22553 122.954 7.64634C125.775 7.14541 128.843 7.60459 131.207 9.35783C132.292 10.0962 133.179 11.0928 133.995 12.1286C134.056 8.52322 133.94 4.91239 134.05 1.312ZM123.043 9.7964C120.668 10.4147 118.541 12.0714 117.388 14.3282C116.343 16.1858 116.129 18.4373 116.486 20.5271C117.051 23.5588 119.247 26.1706 122.022 27.2507C123.821 27.9082 125.823 28.0804 127.67 27.5221C129.524 26.8072 131.262 25.6042 132.367 23.8695C133.769 21.8107 134.281 19.1183 133.706 16.6661C133.133 13.9215 131.219 11.5525 128.78 10.3706C127.004 9.50419 124.947 9.37374 123.043 9.7964ZM109.211 1.38766C110.288 1.04066 111.374 2.48865 110.713 3.47484C110.283 4.30189 109.06 4.48451 108.495 3.70965C107.794 2.9687 108.238 1.59116 109.211 1.38766ZM9.94231 7.60485C12.266 7.26308 14.7254 7.48484 16.8255 8.63801C18.4357 9.44419 19.7496 10.7382 20.8722 12.1601C20.9099 10.6887 20.8572 9.2172 20.9325 7.74835C21.5404 7.73009 22.1509 7.73243 22.7613 7.75096C22.8515 14.9961 22.8088 22.2488 22.7814 29.4971C22.1584 29.6612 21.5153 29.6927 20.8921 29.5072C20.8896 28.0778 20.8973 26.6507 20.8795 25.2207C19.1462 27.3757 16.8802 29.2724 14.1546 29.7734C11.7606 30.3004 9.22837 29.8801 7.03505 28.7895C3.89818 27.1826 1.60714 23.9345 1.09467 20.3393C0.858535 18.2466 1.03941 16.0684 1.91864 14.1482C3.3204 10.7539 6.44295 8.22579 9.94231 7.60485ZM9.98752 9.74683C6.87251 10.433 4.22224 13.0002 3.36336 16.1884C2.2829 19.8853 3.82031 24.1588 6.92276 26.2695C9.00529 27.754 11.7309 28.2054 14.175 27.5818C16.9712 26.6582 19.3928 24.378 20.2394 21.4168C21.4578 17.806 20.0058 13.5507 17.0465 11.3409C15.0695 9.74944 12.4167 9.24329 9.98752 9.74683ZM30.2929 9.68943C33.6716 6.70502 39.1154 6.69954 42.4967 9.6816C44.7224 11.5522 46.0789 14.4873 46.0614 17.4642C46.0488 21.4692 46.0639 25.4738 46.0538 29.4788C45.3931 29.6953 44.7073 29.6846 44.0466 29.484C44.0316 25.2992 44.0793 21.1092 44.024 16.9244C43.8306 12.857 40.183 9.37661 36.2566 9.56968C32.3227 9.52793 28.8358 13.1309 28.7454 17.1957C28.6927 21.297 28.7356 25.3981 28.7256 29.4997C28.0646 29.6745 27.3839 29.6953 26.7259 29.4918C26.7207 25.5679 26.7332 21.644 26.7207 17.7175C26.6604 14.6648 27.9993 11.6096 30.2929 9.68943ZM73.6493 16.932C73.697 11.8077 78.2339 7.15585 83.2182 7.48197C83.2433 8.09508 83.3262 8.72124 83.188 9.32652C82.7182 9.71526 82.0475 9.56916 81.4974 9.72569C79.2641 10.1927 77.3222 11.8494 76.3629 13.9862C75.3831 15.9403 75.6193 18.213 75.5866 20.339C75.5512 23.3498 75.6545 26.3684 75.5389 29.3765C74.9913 29.8357 74.2728 29.64 73.6523 29.4994C73.6493 25.3094 73.6518 21.122 73.6493 16.932ZM91.6259 7.66225C94.1933 7.19264 96.9589 7.5005 99.2326 8.88847C101.991 10.5034 103.998 13.4255 104.558 16.6632C104.837 18.7893 104.744 21.0304 103.903 23.0187C102.425 26.731 98.9336 29.5174 95.0748 29.9377C92.4926 30.2145 89.8046 29.5385 87.6817 27.9732C85.3354 26.3454 83.7351 23.697 83.1775 20.8402C82.5947 18.1164 83.2149 15.2179 84.5616 12.8359C86.1018 10.2086 88.7094 8.25449 91.6259 7.66225ZM91.8922 9.77292C87.5839 10.7226 84.4112 15.4109 85.1221 19.9427C85.4534 22.2699 86.6366 24.4487 88.4406 25.8836C90.493 27.7412 93.4721 28.2317 96.0696 27.5795C97.6698 27.0029 99.2474 26.1732 100.38 24.8244C101.93 23.0659 102.784 20.6526 102.677 18.2678C102.528 15.137 100.735 12.1367 98.0544 10.6704C96.2105 9.55115 93.9597 9.33722 91.8922 9.77292ZM108.437 8.36146C109.012 7.73791 109.927 8.05882 110.628 8.23884C110.595 15.0587 110.643 21.8812 110.605 28.7011C110.54 29.6038 109.381 29.1342 108.809 29.2698C108.706 29.1837 108.497 29.0089 108.392 28.9228C108.422 22.0716 108.331 15.2126 108.437 8.36146Z"
        fill="#E1E3DF"
        stroke="#4CDF9D"
        stroke-width="2"
      />
    </svg>
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Aakash Shah
    </h1>
    <div
      class="MuiBox-root css-13m36ks"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        React & Android Developer
      </h6>
    </div>
  </div>
</div>
`;
